euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/Builders.java

```diff
@@ -106,24 +105,13 @@
      *
      * @return the dataset representing the new operator's output
      */
-    default Dataset<V> outputValues() {
+    default Dataset<V> outputValues(OutputHint... outputHints) {
       return MapElements
           .named("extract-values")
           .of(output())
```

Comment:
The hints should be set here I think.

Reply:
It shouldn't, output Dataset is with values. There would hint get lost. (last operator is MapElements). I added test.

Reply:
Why would the hint get lost? The graph can be arbitrarily complicated and as a general rule, this should yeild the same runtime behavior:
```java
 Dataset input = ...;
 Dataset<Pair> output = ReduceByKey.of(input)
   ...
   .output(hints);
 MapElements.of(input)
   .using(Pair::getSecond)
   .output()
   .persist(...)
```
it should behave the same way as
```java
 Dataset input = ...;
 ReduceByKey.of(input)
    ...
    .outputValues(hints)
    .persist(...)
```
That is because non-shuffle operators should forward hints to downstream Datasets (unless overridden). That is due to the construction of the hints. The operator that the hint applies to is really `RBK`, not the `MapElements`.

Reply:
After discussion, for now `Hints` doesn't propagate downstream in flow. Because we don't know if `Hint` will be still valid in next operator.